[3.1] updated trace_api openapi documentation

ericpassmore opened the pull request:
- added trace_transcation method
- updated return schema for both get_block and get_transaction_trace
     - New v2 schema for BLockTrace added to options for returned schema
     - New v2 and v3 schema for TransactionTrace added to options for returned schema
     - New sub-schemas ActionTraceV2 and BlockTRaceV2
- added new component types
   - ERROR_DETAILS

lparisc commented on plugins/trace_api_plugin/trace_api.swagger.yaml:
```diff
@@ -43,12 +47,288 @@ paths:
-                oneOf:
-                  - $ref: "https://eosio.github.io/schemata/v2.0/oas/BlockTraceV0.yaml"
-                  - $ref: "https://eosio.github.io/schemata/v2.0/oas/BlockTraceV1.yaml"
+                $ref: "#/component/schema/BLOCK_TRACE"
```
This adds v2 block trace support, but removes v0 and v1 support. These should still be supported - see https://github.com/AntelopeIO/leap/blob/release/3.1/plugins/trace_api_plugin/include/eosio/trace_api/trace.hpp

ericpassmore replied:
Good point, added all the BlockTraces, and created a new BlockTraceV3 sub-schema

lparisc commented on plugins/trace_api_plugin/trace_api.swagger.yaml:
```diff
@@ -1,12 +1,13 @@
 openapi: 3.0.0
 info:
   title: Trace API
+  description: This plubing must specifiy one option `trace-rpc-abi` or `no-trace-abis`. These options are provided at nodeos startup on the command line or in config.ini file. Configuration option `trace-rpc-abi` There must be at least one ABI specified OR the flag trace-no-abis must be used. ABIs are specified as "Key=Value" pairs in the form <account-name>=<abi-def> Where <abi-def> can be an a ...
```
This repeats the same description of the `--trace-rpc-abi` option in the [trace api docs](https://docs.eosnetwork.com/leap/latest/nodeos/plugins/trace_api_plugin/#configuration-options). We can just link to the specific configuration option if needed to avoid maintaining duplicate sources of truth.

ericpassmore commented:
Updated get_black and get_transaction_trace to support the multiple return schemas. Updated the sub-schemas as needed to support BlockTraceV3 and TransactionTraceV3. 
Re-synced from release/3.1 branch to support merge

lparisc reviewed (approved):
Nice
